client/signup/steps/emails/test/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import EmailSignupTitanCard from '../index.jsx';
+
+const initialState = {
+	flowName: 'test:flow',
+	stepName: 'test:step2',
+	sites: {
+		items: {
+			1: {},
+		},
+	},
+	ui: {},
+	productsList: {
+		isFetching: false,
+	},
+};
+
+describe( 'Email Step Titan Signup Card', () => {
+	test( 'should display price information', () => {
+		const newInitialState = {
+			...initialState,
+			sites: {
+				...initialState.sites,
+				items: {
+					1: {
+						...initialState.sites.items[ 1 ],
+					},
+				},
+			},
+		};
+		const middlewares = [ thunk ];
+		const mockStore = configureStore( middlewares );
+		const store = mockStore( newInitialState );
+
+		render(
+			<Provider store={ store }>
+				<EmailSignupTitanCard { ...newInitialState } />
+			</Provider>
+		);
+
+		expect( screen.getByText( 'Add' ) ).toBeInTheDocument();
+		expect( screen.queryByText( 'Loading...' ) ).toBeNull();
```

Comment:
```suggestion
		expect( screen.queryByText( 'Loading…' ) ).toBeNull();
```
Adding a test is a great idea, but it's currently passing because we're looking for the wrong string. 

I'm not sure what would be a practical way to test this… Ultimately, I guess we want to assert that somewhere in the component tree, a network request for `public-api.wordpress.com/rest/v1.1/products` was made with the appropriate `type` parameter (either no parameter or `email`).

@nightnei, do you have an idea for how we could test this?

Reply:
Nice catch on the ellipsis, thank you.

> I'm not sure what would be a practical way to test this…

Yeah, after I realized why this happened in the first place, shoring up against future updates to prior onboarding steps felt kind of essential, but writing an effective test was tricky. The actual pricing string feels too brittle and the 'loading' text, while the clearest symptom of the breakage this time, is really just a side effect of what we actually want to know 🤔

Reply:
Quick update to my approach here (PR description has been updated accordingly). While digging into this test further I actually found that the `type='titan'` I'd originally thought we could request doesn't actually exist. Titan products don't have a `productType` value, though it looked to me like they did initially. In the end, the only way to get titan products at the moment is to request _all_ products. Maybe we can optimize that in a followup at some point. 

Now that you've pointed out that incorrect string, I found that this kind of test isn't going to work well at all... we're mocking state, which is where the products are coming from any way. 

I started to try and think of was we could create some example test data and make sure it retrieves the right set, but in the end that us leads back to the point you've made above:

> Ultimately, I guess we want to assert that somewhere in the component tree, a network request for public-api.wordpress.com/rest/v1.1/products was made with the appropriate type parameter (either no parameter or email).

What we really want to test here is that the component is requesting the right product data. With that in mind, I've pushed an updated version of the test, which uses a spy on the `useQueryProductsList` component to inspect its props. This way we can assert:

1. `UseQueryProductsList` is being called when the email step renders
2. It's not passing any `type` argument, so we can be confident we'll get all products, including email.

This feels more reliable, and like it's actually testing the right thing instead of an implementation detail like the lack of `Loading…`. What do you think @fredrikekelund?

Reply:
Looks like a sound way to test this 👍